mathics/builtin/numbers/calculus.py

```diff
@@ -1947,82 +1744,38 @@ class NIntegrate(Builtin):
         "Automatic": (None, False),
     }
 
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-        self.methods["Internal"] = (_internal_adaptative_simpsons_rule, False)
-        try:
-            from scipy.integrate import romberg, quad, nquad
+    try:
+        from mathics.algorithm.integrators import (
+            integrator_methods,
+            integrator_messages,
+        )
 
-            self.methods["NQuadrature"] = (
-                _scipy_interface(
-                    nquad, {}, {"full_output": 1}, lambda res: (res[0], res[1])
-                ),
-                True,
-            )
-            self.methods["Quadrature"] = (
-                _scipy_interface(
-                    quad,
-                    {
-                        "tol": ("epsabs", None),
-                        "maxrec": ("limit", lambda maxrec: int(2 ** maxrec)),
-                    },
-                    {"full_output": 1},
-                    lambda res: (res[0], res[1]),
-                ),
-                False,
-            )
-            self.methods["Romberg"] = (
-                _scipy_interface(
-                    romberg,
-                    {"tol": ("tol", None), "maxrec": ("divmax", None)},
-                    None,
-                    lambda x: (x, np.nan),
-                ),
-                False,
-            )
-            self.methods["Automatic"] = self.methods["Quadrature"]
-        except Exception:
-            self.methods["Automatic"] = self.methods["Internal"]
-            self.methods["Simpson"] = self.methods["Internal"]
+        methods.update(integrator_methods)
+        messages.update(integrator_messages)
+    except Exception:
+        pass
 
-        self.messages["bdmtd"] = (
-            "The Method option should be a "
-            + "built-in method name in {`"
-            + "`, `".join(list(self.methods))
-            + "`}. Using `Automatic`"
+    try:
+        from mathics.builtin.scipy_utils.integrators import (
+            scipy_nintegrate_methods,
+            scipy_nintegrate_messages,
```

Comment:
Here I have a doubt: to have the examples in the doctests, we need to install scipy. One possibility is to modify the docstring of `NIntegrate` according to the scipy availability. Another possibility is to move all of this to a pymathics module that requires scipy, and then load it optionally. Also, I can remove the examples or add the dependency to the Github actions. Thoughts?